recipe/meta.yaml

```diff
@@ -35,7 +35,13 @@ requirements:
 
 test:
   commands:
-    - oommf +version  # [unix]
+    - conda install -y libx11-common-cos6-x86_64  # [linux]
+    - conda install -y libx11-cos6-x86_64  # [linux]
+    - conda install -y libx11-devel-cos6-x86_64  # [linux]
+    - conda install -y libxau-cos6-x86_64  # [linux]
+    - conda install -y libxau-cos6-x86_64  # [linux]
+    - conda install -y xorg-x11-proto-devel-cos6-x86_64  # [linux]
```

Comment:
You should not install anything here in the test phase, if these things are dependencies of the package it should be explicitly declared in the run requirements using CDT packages

Reply:
Basically you have to add 
```
    - {{ cdt('xorg-x11-proto-devel') }}  # [linux]
    - {{ cdt('libx11-devel') }}  # [linux]
    - {{ cdt('libxau') }}  # [linux]
    - {{ cdt('libxcb') }}  # [linux]
```
in run requirements as well

Reply:
We have this dependencies on the build stage:

    - {{ cdt('xorg-x11-proto-devel') }}  # [linux]
    - {{ cdt('libx11-devel') }}  # [linux]
    - {{ cdt('libxau') }}  # [linux]
    - {{ cdt('libxcb') }}  # [linux]

But it is not work. I added this line on the test phase, try to solve this problem with `libX11.so.6`

Reply:
If it is complaining about ``libX11`` that shows this is a dependency at run time as well.
You will need to add at least
```
- {{ cdt('libx11-devel') }}  # [linux]
```

Reply:
The same problem:

```
+ oommf boxsi +fg /home/conda/feedstock_root/build_artifacts/oommf_1550663346314/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/opt/oommf/app/oxs/examples/stdprob4.mif -exitondone 1
/home/conda/feedstock_root/build_artifacts/oommf_1550663346314/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/opt/oommf/app/oxs/linux-x86_64/oxs: error while loading shared libraries: libX11.so.6: cannot open shared object file: No such file or directory
<1282> oommf.tcl 2.0a1  panic:
child process exited abnormally
```

Reply:
Probably you have to add
```
- {{ cdt('xorg-x11-proto-devel') }}  # [linux]
```
as well

Reply:
We tried to add all CDT packages to `host` and `run` section - the same problem.
For me it is looks like, when conda combine our packages, it add `.so` library only for tk, zlib, compiler, but not for CDT packages:

```
INFO:conda_build.build:Packaging oommf
Packaging oommf
Packaging oommf-2.0a1-h2861bca_1001
INFO:conda_build.build:Packaging oommf-2.0a1-h2861bca_1001
number of files: 1535
Warning: rpath /home/conda/feedstock_root/build_artifacts/oommf_1550663346314/_build_env/lib is outside prefix /home/conda/feedstock_root/build_artifacts/oommf_1550663346314/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place (removing it)
patchelf: file: /home/conda/feedstock_root/build_artifacts/oommf_1550663346314/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/opt/oommf/app/mmdisp/linux-x86_64/mmdispsh
    setting rpath to: $ORIGIN/../../../../../lib
Warning: rpath /home/conda/feedstock_root/build_artifacts/oommf_1550663346314/_build_env/lib is outside prefix /home/conda/feedstock_root/build_artifacts/oommf_1550663346314/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place (removing it)
patchelf: file: /home/conda/feedstock_root/build_artifacts/oommf_1550663346314/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/opt/oommf/pkg/oc/linux-x86_64/varinfo
    setting rpath to: $ORIGIN/../../../../../lib
Warning: rpath /home/conda/feedstock_root/build_artifacts/oommf_1550663346314/_build_env/lib is outside prefix /home/conda/feedstock_root/build_artifacts/oommf_1550663346314/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place (removing it)
patchelf: file: /home/conda/feedstock_root/build_artifacts/oommf_1550663346314/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/opt/oommf/app/mmsolve/linux-x86_64/mmsolve
    setting rpath to: $ORIGIN/../../../../../lib
Warning: rpath /home/conda/feedstock_root/build_artifacts/oommf_1550663346314/_build_env/lib is outside prefix /home/conda/feedstock_root/build_artifacts/oommf_1550663346314/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place (removing it)
patchelf: file: /home/conda/feedstock_root/build_artifacts/oommf_1550663346314/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/opt/oommf/app/omfsh/linux-x86_64/omfsh
    setting rpath to: $ORIGIN/../../../../../lib
Warning: rpath /home/conda/feedstock_root/build_artifacts/oommf_1550663346314/_build_env/lib is outside prefix /home/conda/feedstock_root/build_artifacts/oommf_1550663346314/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place (removing it)
patchelf: file: /home/conda/feedstock_root/build_artifacts/oommf_1550663346314/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/opt/oommf/app/mmdisp/linux-x86_64/condispsh
    setting rpath to: $ORIGIN/../../../../../lib
Warning: rpath /home/conda/feedstock_root/build_artifacts/oommf_1550663346314/_build_env/lib is outside prefix /home/conda/feedstock_root/build_artifacts/oommf_1550663346314/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place (removing it)
patchelf: file: /home/conda/feedstock_root/build_artifacts/oommf_1550663346314/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/opt/oommf/app/oxs/linux-x86_64/oxs
    setting rpath to: $ORIGIN/../../../../../lib
Warning: rpath /home/conda/feedstock_root/build_artifacts/oommf_1550663346314/_build_env/lib is outside prefix /home/conda/feedstock_root/build_artifacts/oommf_1550663346314/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place (removing it)
patchelf: file: /home/conda/feedstock_root/build_artifacts/oommf_1550663346314/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/opt/oommf/app/omfsh/linux-x86_64/filtersh
    setting rpath to: $ORIGIN/../../../../../lib
   INFO (oommf,opt/oommf/app/mmdisp/linux-x86_64/mmdispsh): Needed DSO x86_64-conda_cos6-linux-gnu/sysroot/lib/libm.so.6 found in CDT/compiler package conda-forge::gcc_impl_linux-64-7.3.0-habb00fd_1
   INFO (oommf,opt/oommf/app/mmdisp/linux-x86_64/mmdispsh): Needed DSO x86_64-conda_cos6-linux-gnu/sysroot/lib/libc.so.6 found in CDT/compiler package conda-forge::gcc_impl_linux-64-7.3.0-habb00fd_1
   INFO (oommf,opt/oommf/app/mmdisp/linux-x86_64/mmdispsh): Needed DSO lib/libstdc++.so.6 found in conda-forge::libstdcxx-ng-7.3.0-hdf63c60_0
   INFO (oommf,opt/oommf/app/mmdisp/linux-x86_64/mmdispsh): Needed DSO lib/libtcl8.6.so found in conda-forge::tk-8.6.9-h84994c4_1000
   INFO (oommf,opt/oommf/app/mmdisp/linux-x86_64/mmdispsh): Needed DSO lib/libz.so.1 found in conda-forge::zlib-1.2.11-h14c3975_1004
   INFO (oommf,opt/oommf/app/mmdisp/linux-x86_64/mmdispsh): Needed DSO lib/libtk8.6.so found in conda-forge::tk-8.6.9-h84994c4_1000
   INFO (oommf,opt/oommf/app/mmdisp/linux-x86_64/mmdispsh): Needed DSO lib/libgcc_s.so.1 found in conda-forge::libgcc-ng-7.3.0-hdf63c60_0
   INFO (oommf,opt/oommf/pkg/oc/linux-x86_64/varinfo): Needed DSO x86_64-conda_cos6-linux-gnu/sysroot/lib/libm.so.6 found in CDT/compiler package conda-forge::gcc_impl_linux-64-7.3.0-habb00fd_1
   INFO (oommf,opt/oommf/pkg/oc/linux-x86_64/varinfo): Needed DSO x86_64-conda_cos6-linux-gnu/sysroot/lib/libc.so.6 found in CDT/compiler package conda-forge::gcc_impl_linux-64-7.3.0-habb00fd_1
   INFO (oommf,opt/oommf/pkg/oc/linux-x86_64/varinfo): Needed DSO lib/libz.so.1 found in conda-forge::zlib-1.2.11-h14c3975_1004
   INFO (oommf,opt/oommf/app/mmsolve/linux-x86_64/mmsolve): Needed DSO x86_64-conda_cos6-linux-gnu/sysroot/lib/libm.so.6 found in CDT/compiler package conda-forge::gcc_impl_linux-64-7.3.0-habb00fd_1
   INFO (oommf,opt/oommf/app/mmsolve/linux-x86_64/mmsolve): Needed DSO x86_64-conda_cos6-linux-gnu/sysroot/lib/libc.so.6 found in CDT/compiler package conda-forge::gcc_impl_linux-64-7.3.0-habb00fd_1
   INFO (oommf,opt/oommf/app/mmsolve/linux-x86_64/mmsolve): Needed DSO lib/libgcc_s.so.1 found in conda-forge::libgcc-ng-7.3.0-hdf63c60_0
   INFO (oommf,opt/oommf/app/mmsolve/linux-x86_64/mmsolve): Needed DSO lib/libz.so.1 found in conda-forge::zlib-1.2.11-h14c3975_1004
   INFO (oommf,opt/oommf/app/mmsolve/linux-x86_64/mmsolve): Needed DSO lib/libtcl8.6.so found in conda-forge::tk-8.6.9-h84994c4_1000
   INFO (oommf,opt/oommf/app/mmsolve/linux-x86_64/mmsolve): Needed DSO lib/libtk8.6.so found in conda-forge::tk-8.6.9-h84994c4_1000
   INFO (oommf,opt/oommf/app/mmsolve/linux-x86_64/mmsolve): Needed DSO lib/libstdc++.so.6 found in conda-forge::libstdcxx-ng-7.3.0-hdf63c60_0
   INFO (oommf,opt/oommf/app/omfsh/linux-x86_64/omfsh): Needed DSO x86_64-conda_cos6-linux-gnu/sysroot/lib/libm.so.6 found in CDT/compiler package conda-forge::gcc_impl_linux-64-7.3.0-habb00fd_1
   INFO (oommf,opt/oommf/app/omfsh/linux-x86_64/omfsh): Needed DSO x86_64-conda_cos6-linux-gnu/sysroot/lib/libc.so.6 found in CDT/compiler package conda-forge::gcc_impl_linux-64-7.3.0-habb00fd_1
   INFO (oommf,opt/oommf/app/omfsh/linux-x86_64/omfsh): Needed DSO lib/libstdc++.so.6 found in conda-forge::libstdcxx-ng-7.3.0-hdf63c60_0
   INFO (oommf,opt/oommf/app/omfsh/linux-x86_64/omfsh): Needed DSO lib/libz.so.1 found in conda-forge::zlib-1.2.11-h14c3975_1004
   INFO (oommf,opt/oommf/app/omfsh/linux-x86_64/omfsh): Needed DSO lib/libgcc_s.so.1 found in conda-forge::libgcc-ng-7.3.0-hdf63c60_0
   INFO (oommf,opt/oommf/app/omfsh/linux-x86_64/omfsh): Needed DSO lib/libtk8.6.so found in conda-forge::tk-8.6.9-h84994c4_1000
   INFO (oommf,opt/oommf/app/omfsh/linux-x86_64/omfsh): Needed DSO lib/libtcl8.6.so found in conda-forge::tk-8.6.9-h84994c4_1000
   INFO (oommf,opt/oommf/app/mmdisp/linux-x86_64/condispsh): Needed DSO x86_64-conda_cos6-linux-gnu/sysroot/lib/libm.so.6 found in CDT/compiler package conda-forge::gcc_impl_linux-64-7.3.0-habb00fd_1
   INFO (oommf,opt/oommf/app/mmdisp/linux-x86_64/condispsh): Needed DSO x86_64-conda_cos6-linux-gnu/sysroot/lib/libc.so.6 found in CDT/compiler package conda-forge::gcc_impl_linux-64-7.3.0-habb00fd_1
   INFO (oommf,opt/oommf/app/mmdisp/linux-x86_64/condispsh): Needed DSO lib/libstdc++.so.6 found in conda-forge::libstdcxx-ng-7.3.0-hdf63c60_0
   INFO (oommf,opt/oommf/app/mmdisp/linux-x86_64/condispsh): Needed DSO lib/libtcl8.6.so found in conda-forge::tk-8.6.9-h84994c4_1000
   INFO (oommf,opt/oommf/app/mmdisp/linux-x86_64/condispsh): Needed DSO lib/libz.so.1 found in conda-forge::zlib-1.2.11-h14c3975_1004
   INFO (oommf,opt/oommf/app/mmdisp/linux-x86_64/condispsh): Needed DSO lib/libtk8.6.so found in conda-forge::tk-8.6.9-h84994c4_1000
   INFO (oommf,opt/oommf/app/mmdisp/linux-x86_64/condispsh): Needed DSO lib/libgcc_s.so.1 found in conda-forge::libgcc-ng-7.3.0-hdf63c60_0
   INFO (oommf,opt/oommf/app/oxs/linux-x86_64/oxs): Needed DSO x86_64-conda_cos6-linux-gnu/sysroot/lib/libm.so.6 found in CDT/compiler package conda-forge::gcc_impl_linux-64-7.3.0-habb00fd_1
   INFO (oommf,opt/oommf/app/oxs/linux-x86_64/oxs): Needed DSO lib/libz.so.1 found in conda-forge::zlib-1.2.11-h14c3975_1004
   INFO (oommf,opt/oommf/app/oxs/linux-x86_64/oxs): Needed DSO x86_64-conda_cos6-linux-gnu/sysroot/lib/libc.so.6 found in CDT/compiler package conda-forge::gcc_impl_linux-64-7.3.0-habb00fd_1
   INFO (oommf,opt/oommf/app/oxs/linux-x86_64/oxs): Needed DSO lib/libtcl8.6.so found in conda-forge::tk-8.6.9-h84994c4_1000
   INFO (oommf,opt/oommf/app/oxs/linux-x86_64/oxs): Needed DSO lib/libtk8.6.so found in conda-forge::tk-8.6.9-h84994c4_1000
   INFO (oommf,opt/oommf/app/oxs/linux-x86_64/oxs): Needed DSO x86_64-conda_cos6-linux-gnu/sysroot/lib/librt.so.1 found in CDT/compiler package conda-forge::gcc_impl_linux-64-7.3.0-habb00fd_1
   INFO (oommf,opt/oommf/app/oxs/linux-x86_64/oxs): Needed DSO lib/libgcc_s.so.1 found in conda-forge::libgcc-ng-7.3.0-hdf63c60_0
   INFO (oommf,opt/oommf/app/oxs/linux-x86_64/oxs): Needed DSO x86_64-conda_cos6-linux-gnu/sysroot/lib/ld-linux-x86-64.so.2 found in CDT/compiler package conda-forge::gcc_impl_linux-64-7.3.0-habb00fd_1
   INFO (oommf,opt/oommf/app/oxs/linux-x86_64/oxs): Needed DSO lib/libstdc++.so.6 found in conda-forge::libstdcxx-ng-7.3.0-hdf63c60_0
   INFO (oommf,opt/oommf/app/omfsh/linux-x86_64/filtersh): Needed DSO x86_64-conda_cos6-linux-gnu/sysroot/lib/libm.so.6 found in CDT/compiler package conda-forge::gcc_impl_linux-64-7.3.0-habb00fd_1
   INFO (oommf,opt/oommf/app/omfsh/linux-x86_64/filtersh): Needed DSO x86_64-conda_cos6-linux-gnu/sysroot/lib/libc.so.6 found in CDT/compiler package conda-forge::gcc_impl_linux-64-7.3.0-habb00fd_1
   INFO (oommf,opt/oommf/app/omfsh/linux-x86_64/filtersh): Needed DSO lib/libstdc++.so.6 found in conda-forge::libstdcxx-ng-7.3.0-hdf63c60_0
   INFO (oommf,opt/oommf/app/omfsh/linux-x86_64/filtersh): Needed DSO lib/libz.so.1 found in conda-forge::zlib-1.2.11-h14c3975_1004
   INFO (oommf,opt/oommf/app/omfsh/linux-x86_64/filtersh): Needed DSO lib/libgcc_s.so.1 found in conda-forge::libgcc-ng-7.3.0-hdf63c60_0
   INFO (oommf,opt/oommf/app/omfsh/linux-x86_64/filtersh): Needed DSO lib/libtk8.6.so found in conda-forge::tk-8.6.9-h84994c4_1000
   INFO (oommf,opt/oommf/app/omfsh/linux-x86_64/filtersh): Needed DSO lib/libtcl8.6.so found in conda-forge::tk-8.6.9-h84994c4_1000
Fixing permissions
Packaged license file.
Detected hard-coded path in text file bin/oommf
Compressing to /tmp/tmp8vab5m_j/oommf-2.0a1-h2861bca_1001.tar.bz2
```

Reply:
Please add this dependency to your run requirements ``xorg-libx11  # [linux]``

Reply:
Fantastic, you are a genius :)

Reply:
If you have any problems in the future just ping me or the ``conda-forge/help-c-cpp`` (for C/C++ projects)